router/Route.js

```diff
@@ -341,11 +341,30 @@ export default class Route {
 	_substituteOptionalParamInPath(path, paramName, paramValue) {
 		return path.replace(
 			new RegExp(`(^|\/):\\\?${paramName}([\/\?]|$)`),
-			paramValue ? '$1' + paramValue +  '$2' : ''
+			paramValue ? '$1' + this._encodePathParamValue(paramValue) +  '$2' : ''
 		);
 	}
 
 	/**
+	 * Encode special characters of path parameter value.
+	 * 
+	 * @param {string|number} paramValue Path parameter value
+	 * @return {string} Encoded path parameter value.
+	 */
+	_encodePathParamValue(paramValue) {
+		// Percen sing (%) has to be first!
+		const charsToEncode = ['%', '/', '?', '#', ':'];
+
+		let newParamValue = String(paramValue);
```

Comment:
1. Calling `String` without `new` looks like an error at the first glance, that's why the linter complains as well :). Please use some other form of conversion, such as: `${paramValue}`.

2. `newParamValue` does not communicate the purpose of the variable, please use names such as `escapedParamValue` instead